server/utils/constants.ts

```diff
@@ -69,6 +69,7 @@ export enum DETECTOR_STATE {
   FEATURE_REQUIRED = 'Feature required',
   INIT_FAILURE = 'Initialization failure',
   UNEXPECTED_FAILURE = 'Unexpected failure',
+  NO_DATA = 'No data',
```

Comment:
Not related with this PR. But we may need to test more edge cases like not enough data, exceed our max running AD task limit etc.

Reply:
Yeah, agreed. I think this "no data" case can be pretty common, so definitely good to have it separated here, but definitely doesn't cover all cases that would currently fall under the 'Unexpected failure' state. I'm assuming those other cases would fall under the `FAILED` state from the backend, and it's probably a matter of parsing out the error message to categorize into other states similar to what's done here.

Reply:
Yes, we can also parse the error message for "FAILED" state. But not so easy for exceptions which has no readable error message.